src/file_pipe_log/pipe.rs

```diff
@@ -393,17 +543,32 @@ mod tests {
     use super::super::format::LogFileFormat;
     use super::super::pipe_builder::lock_dir;
     use super::*;
-    use crate::env::DefaultFileSystem;
+    use crate::env::{DefaultFileSystem, WriteExt};
     use crate::util::ReadableSize;
+    use std::io::{Read, Seek, SeekFrom, Write};
 
     fn new_test_pipe(cfg: &Config, queue: LogQueue) -> Result<SinglePipe<DefaultFileSystem>> {
+        let capacity_of_recycle = |default_file_size: usize, purge_threshold: usize| -> usize {
```

Comment:
We have gone over this before. Only use closure if it's necessary. Function is more than enough here.